server/core/runtime/post_workflow_hook_runner.go

```diff
@@ -47,8 +54,25 @@ func (wh DefaultPostWorkflowHookRunner) Run(ctx models.WorkflowHookCommandContex
 	if err != nil {
 		err = fmt.Errorf("%s: running %q in %q: \n%s", err, command, path, out)
 		ctx.Log.Debug("error: %s", err)
-		return "", err
+		return "", "", err
+	}
+
+	// Read the value from the "outputFilePath" file
+	// to be returned as a custom description.
+	var customStatusOut []byte
+	if _, err := os.Stat(outputFilePath); err == nil {
```

Comment:
If there is an err, then we should handle that, right? At the very least maybe log something to debug

Reply:
We can let it silently fail. It basically means that the user didn't have a custom description to set as part of the workflow since the file doesn't exist.

Reply:
Why does the description have to be saved as an output file? Shouldn't we pass the description to the function instead of outputting a file on the disk and then reading it later?

Reply:
I'm using a file because right now the stdout is used for the details link. You gave the idea of a regex [here](https://github.com/runatlantis/atlantis/pull/2441#issuecomment-1345742992) but I believe that this would make testing kinda hard for users, so that's why I've added the file.

https://github.com/runatlantis/atlantis/blob/e176af2a0c5bff36a70390f183f710cde45bed15/runatlantis.io/docs/post-workflow-hooks.md?plain=1#L79

Reply:
Do you believe there's a better way of handling custom status? I've tried setting an environment variable but that didn't work, the [Cmd struct](https://pkg.go.dev/os/exec#Cmd) has no way of returning an environment variable, you can only set but not get from the process.

Reply:
Hmm, I just noticed that I can get the pid from the Cmd, that means that I probably can try to read `/proc/$pid/environ`, maybe that can enable the usage of environment variables?

Reply:
Let's do a file for now since it works.